e2e/tests/accessSegregation_test.js

```diff
@@ -2,51 +2,62 @@ const config = require('../config.js');
 
 let caseId;
 
-Feature('Cases visible only to respective local authority and admin');
+Feature('Access segregation');
 
 Before(async (I, caseViewPage, submitApplicationEventPage) => {
-  I.logInAndCreateCase(config.swanseaLocalAuthorityEmailUserOne, config.localAuthorityPassword);
-  caseId = await I.grabTextFrom('.heading-h1');
-  await I.enterMandatoryFields();
-  caseViewPage.goToNewActions(config.applicationActions.submitCase);
-  submitApplicationEventPage.giveConsent();
-  I.continueAndSubmit();
-  I.wait(2); // in seconds; time needed for access grant calls to complete
-  I.signOut();
+  if (!caseId) {
+    await I.logInAndCreateCase(config.swanseaLocalAuthorityEmailUserOne, config.localAuthorityPassword);
+    await I.enterMandatoryFields();
+    caseViewPage.goToNewActions(config.applicationActions.submitCase);
+    submitApplicationEventPage.giveConsent();
+    I.continueAndSubmit();
+
+    // eslint-disable-next-line require-atomic-updates
+    caseId = await I.grabTextFrom('.heading-h1');
```

Comment:
Hmm, what about reading caseId from URL? I guess two options are not perfect, but maybe URL more consistent?

Reply:
Not a good reason for change for me. Reading heading worked well from the beginning.